Домашнее задание 4

vmikulitskii opened the pull request.

vmikulitskii committed:
DZ4

DmitryChitalov commented on Урок 4. Практическое задание/task_1.py:
```diff
@@ -12,3 +12,156 @@
+Генераторы должны выполняться быстрее, чем аналогичный код с циклом for,
+но в данном примере получилось немного дольше. Пробовал много раз, 9 из 10
+первый код выполняется быстрее. Возможно потому что в первом вывод результата идет через переменную,
+а во втором создается дополнительный массив, из-за которого время увеличивается.
```
плюс за аналитику
но
не увидел оценки сложности, хотя это есть в задании

DmitryChitalov commented on Урок 4. Практическое задание/task_1.py:
```diff
@@ -12,3 +12,156 @@
+22.766292425
+19.144336706999997
+В данной задаче цикл справляется быстрее, чем рекурсия.
+"""
```
почему цикл быстрее
как считаете это достаточная аналитика?
наша цель обосновать все выводы

DmitryChitalov commented on Урок 4. Практическое задание/task_2.py:
```diff
@@ -12,3 +12,74 @@
+
+Вывод: Решето Эратосфена дает ощутимый прирост скорости.
+Вывод №2 первый алгоритм на столько плох, что не дает выигрыш даже при небольших числах.
+"""
```
всегда ли Решето быстрее??
не увидел оценки сложности, хотя это есть в задании

vmikulitskii committed:
Изменил 1-й скрипт

Теперь скрипт по перебору как и должен работает быстрее решета, до 2000-2500 элемента, далее решето работает быстрее
1) при i = 35
0.0005572279999999999
0.0015637590000000048
2) при i = 2500
6.466714683999999
6.577390780000001
3) при i = 3500
13.609293358999999
12.913102282999999
